### DISCUSSION TITLE
Complejidades extras a los eventos

### DISCUSSION THREAD

**Discussion by vnloyola** [IIC2133-PUC/2024-2#100] [Tarea 1] [answered]
Hola! Estuve leyendo las demás issues sobre las complejidades de los eventos, pero sigo con la siguiente duda. 

Tengo el código así

```c
if (string_equals(command, "ENTER")) {
      enter(festival, input_file); //  O(log(n))
      Ordenar(festival);
}
```

Donde la función `enter` cumple con la complejidad solicitada y luego de hacer la inserción yo ordeno mi festival para poder utilizarlo en el siguiente evento. La complejidad de `Ordenar` se suma a la de `enter` o son independientes?

**Comment by Segismundios** [comment, marked as answer]
Hola @vnloyola, disculpa la demora. En el caso que mencionas, la complejidad que aporta "Ordenar" también sería considerada como parte del evento. Se considerará todo proceso que sea desencadenado por el evento.
Saludos!

**Reply by vnloyola** [reply]
Pero por ejemplo si coloco la función fuera del if de ENTER, se seguirá considerando? Es decir, luego de que se inserte el elemento en el heap. Es que mi función ordenar tiene la finalidad de aplicar HeapSort y poder cumplir con las complejidades de los eventos siguientes.

En ese caso podría ponerla antes de cada evento?

Muchas gracias

**Reply by Segismundios** [reply]
Claro, igualmente contaría como parte de la complejidad del evento. La idea es que luego del enter, todos los ajustes necesarios para que el resto de eventos sigan funcionando tengan complejidad de O(log(n)) a lo más. Si se pusiera fuera del if, esa complejidad se estaría sumando a todos los eventos, porque se aplicaría en todos los ciclos del for.
